crates/pallet-domains/src/lib.rs

```diff
@@ -684,6 +685,18 @@ mod pallet {
     #[pallet::storage]
     pub(super) type AccumulatedTreasuryFunds<T> = StorageValue<_, BalanceOf<T>, ValueQuery>;
 
+    /// Storage used to keep track of which consensus block the domain runtime upgrade happen.
+    ///
+    /// TODO: determine and clear the unneeded domain runtime upgrade record
```

Comment:
not sure I understand what this TODO here. We are ideally need to clear this specific upgrade in consensus as soon as we verify the ER. What am I missing ?

Reply:
Shuldn't we be able to remove the storage with reference count. Since we would know how many domains instantiated with given time ?

Reply:
The last commit adds reference count and removes the upgrade record when it is unneeded, PTAL